ambari-server/src/main/java/org/apache/ambari/server/controller/ServiceGroupRequest.java

```diff
@@ -23,10 +23,12 @@
 
   private String clusterName; // REF
   private String serviceGroupName; // GET/CREATE/UPDATE/DELETE
+  private String version; // Associated stack version info
 
-  public ServiceGroupRequest(String clusterName, String serviceGroupName) {
+  public ServiceGroupRequest(String clusterName, String serviceGroupName, String version) {
     this.clusterName = clusterName;
```

Comment:
I think there's still a misunderstanding here - I thought it would be cleaner to separate this out into 2 fields so that it's clear to the user what is expected: mpack name and then mpack version. You can combine this and call it a stack ID internally, that's fine - but we should take 2 different properties.

Same goes for API requests where we're currently given `name-version` as a string. This should be broken out into 2 fields.

Reply:
I am ok for this change.

Reply:
@jayush What do you think?

Reply:
Lets keep it as "version" for now for legacy purposes. When we upgrade to ambari 3.0 assuming the existing cluster is on HDP-2.6 stack, calling this as mpack name and mpack version would be misleading. 

We should handle this in a separate JIRA if we really want to change this convention for all APIs.

Reply:
But "version" in legacy terms meant only the numeric value. If we don't want to break it out into 2 references, I think we'd rather use stackId, right?